README-PROVIDERS.md

```diff
@@ -0,0 +1,154 @@
+This may mean that some applications upgrading from earlier versions of OpenSSL
+may find that some algorithms are no longer available unless they load the
+legacy provider explicitly.
+
+Algorithms in the legacy provider include MD2, MD4, MDC2, RMD160, CAST5,
+BF (Blowfish), IDEA, SEED, RC2, RC4, RC5 and DES (but not 3DES).
+
+The FIPS Provider
+-----------------
+
+The FIPS provider contains a sub-set of the algorithm implementations available
+from the default provider, consisting of algorithms conforming to FIPS standards.
+It is intended that this provider will be FIPS140-2 validated.
+
+In some cases there may be minor behavioural differences between algorithm
+implementations in this provider compared to the equivalent algorithm in the
+default provider. This is typically in order to conform to FIPS standards.
+
+The Base Provider
+-----------------
+
+The base provider contains a small sub-set of non-cryptographic algorithms
+available in the default provider. For example, it contains algorithms to
+serialize and deserialize keys to files. If you do not load the default
+provider then you should always load this one instead (in particular, if
+you are using the FIPS provider).
+
+The Null Provider
+-----------------
+
+The null provider is "built-in" to libcrypto and contains no algorithm
+implementations. In order to guarantee that the default provider is not
+automatically loaded, the null provider can be loaded instead.
+
+This can be useful if you are using non-default library contexts and want
+to ensure that the default library context is never used unintentionally.
+
+
+Loading Providers
+=================
+
+
+Providers to be loaded can be specified in the OpenSSL config file.
+See the [config(5)] manual page for information about how to configure
+providers via the config file, and how to automatically activate them.
+
+ [config(5)]: https://www.openssl.org/docs/manmaster/man5/config.html
+
+The following is a minimal config file example to load and activate both
+the legacy and the default provider in the default library context.
+
+    openssl_conf = openssl_init
+    
+    [openssl_init]
+    providers = provider_sect
+    
+    [provider_sect]
+    default = default_sect
+    legacy = legacy_sect
+    
+    [default_sect]
+    activate = 1
+    
+    [legacy_sect]
+    activate = 1
+    
+
+It is also possible to load providers programmatically. For example you can
+load the legacy provider into the default library context as shown below.
+Note that once you have explicitly loaded a provider into the library context
+the default provider will no longer be automatically loaded. Therefore you will
+often also want to explicitly load the default provider, as is done here:
+
+
+    #include <stdio.h>
+    #include <stdlib.h>
+    
+    #include <openssl/provider.h>
+    
+    int main(void)
+    {
+        OSSL_PROVIDER *legacy;
+        OSSL_PROVIDER *deflt;
+    
```

Comment:
Note: git complains about a lot of trailing whitespace in the empty lines of the code block. I'm not sure yet whether these lines can be left entirely empty, or whether the four spaces are required to keep the code block contiguous.

Reply:
You can try removing the trailing whitespace, push to your branch, and see what it looks like.  I assume you know that.

Reply:
Sure. Just was in a hurry to do some shopping before 8 PM. ;-)

Reply:
I normally just create an empty markdown file via GitHub web interface, paste my contents and switch to preview mode. No need to push anything...

Reply:
Ok, GitHub says "no problem". Whitespace cleanup pushed.